internal/packages/packages.go

```diff
@@ -222,6 +225,68 @@ func ReadPackageManifestFromPackageRoot(packageRoot string) (*PackageManifest, e
 	return ReadPackageManifest(filepath.Join(packageRoot, PackageManifestFile))
 }
 
+// ReadPackageManifestFromZipPackage reads and parses the package manifest file for the given zip package.
+func ReadPackageManifestFromZipPackage(zipPackage string) (*PackageManifest, error) {
```

Comment:
New method to read the Package manifest from a zip file. Needed for two reasons when using a zip file:
- InstalledPackage requires a packages.PackageManifest that it us used in the assets test runner
- `check-condition` parameter requires also the manifest to be able to check the constraints.